### PR TITLE
qisrc push: add --force

### PR DESCRIPTION
**Pull request by Phlogistique**

### CODE REVIEW

File: python/qisrc/test/test_qisrc_push.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def test_not_under_code_review_with_no_review(qisrc_action, git_server):
     (_, remote) = foo_git.call("ls-remote", "origin", "master", raises=False)
     assert remote == "%s\trefs/heads/master" % sha1
 
-def test_using_dash_y(qisrc_action, git_server):
+def test_using_dash_y_and_dash_f(qisrc_action, git_server):
```
Comment (vbarbaresi) [inline python/qisrc/test/test_qisrc_push.py]: I prefer a separate test for dash_f
Good practice is one assert per test, test one thing at a time. It's easier to diagnose failures

**Comment by vbarbaresi** [comment]
Nice one, just one comment on the test
Also I would like a "paranoid" test: check that it doesn't --force when using the classic qisrc push
It may seem like a stupid test but for such a critical feature it's better to be safe

File: python/qisrc/test/test_qisrc_push.py (hunk 1 above)
Comment (Phlogistique) [inline python/qisrc/test/test_qisrc_push.py]: is there a better way to use the test for -y as a precondition for the test of -f?

Comment (vbarbaresi) [inline python/qisrc/test/test_qisrc_push.py]: I don't think so, but it's only 3 lines duplicated, not really a problem in the tests

**Comment by Phlogistique** [comment]
Paranoid test added.

File: python/qisrc/test/test_qisrc_push.py [hunk 2]
```diff
@@ -50,6 +52,56 @@ def test_using_dash_y(qisrc_action, git_server):
...
+    foo_git.commit_file("a.txt", "a")
+    qisrc_action("push",  "--project", "foo", "--no-review")
+    _, sha1 = foo_git.call("log", "-1", "--pretty=%H", raises=False)
+    (_, remote) = foo_git.call("ls-remote", "gerrit", "master", raises=False)
+
+    foo_git.commit("--amend", "-mfoobar")
+    _, new_sha1 = foo_git.call("log", "-1", "--pretty=%H", raises=False)
+
+    with pytest.raises(qisys.command.CommandFailedException):
+      qisrc_action("push",  "--project", "foo", "--no-review")
```
Comment (vbarbaresi) [inline python/qisrc/test/test_qisrc_push.py]: bad indentation, and 2 spaces before ``"--project"``

Comment (Phlogistique) [inline python/qisrc/test/test_qisrc_push.py]: fixed

File: python/qisrc/test/test_qisrc_push.py [hunk 3]
```diff
@@ -50,6 +52,55 @@ def test_using_dash_y(qisrc_action, git_server):
...
+    foo_git = TestGit(foo_proj.path)
+    foo_git.commit_file("a.txt", "a")
+    qisrc_action("push",  "--project", "foo", "-y")
+    _, sha1 = foo_git.call("log", "-1", "--pretty=%H", raises=False)
+    (_, remote) = foo_git.call("ls-remote", "origin", "master", raises=False)
+
+    foo_git.commit("--amend", "-mfoobar")
+    _, new_sha1 = foo_git.call("log", "-1", "--pretty=%H", raises=False)
+
+    with pytest.raises(qisys.command.CommandFailedException):
```
Comment (vbarbaresi) [inline python/qisrc/test/test_qisrc_push.py]: #pylint:disable-msg=E1101

File: python/qisrc/test/test_qisrc_push.py [hunk 4]
```diff
@@ -50,6 +52,55 @@ def test_using_dash_y(qisrc_action, git_server):
...
+    foo_git = TestGit(foo_proj.path)
+    foo_git.commit_file("a.txt", "a")
+    qisrc_action("push",  "--project", "foo", "--no-review")
+    _, sha1 = foo_git.call("log", "-1", "--pretty=%H", raises=False)
+    (_, remote) = foo_git.call("ls-remote", "gerrit", "master", raises=False)
+
+    foo_git.commit("--amend", "-mfoobar")
+    _, new_sha1 = foo_git.call("log", "-1", "--pretty=%H", raises=False)
+
+    with pytest.raises(qisys.command.CommandFailedException):
```
Comment (vbarbaresi) [inline python/qisrc/test/test_qisrc_push.py]: ``# pylint:disable-msg=E1101``

**Comment by vbarbaresi** [comment]
Looks good to me, merging!

**Pull request merged**